tethysapp/hydro_var_monitor/controllers.py

```diff
@@ -215,21 +292,70 @@ def get_plot(request):
     return JsonResponse(json.loads(json.dumps(plot_data)))
 
 
-@controller(name='get_predefined', url='hydro-var-monitor/get-predefined',app_workspace=True)
+@controller(name='admin-upload-exact-zipped-data', url='hydro-var-monitor/admin/unzip-exact', app_workspace=True)
```

Comment:
this should require the user to be logged in and be admin. See tethys docs for how to use the @login_required decorator and another decorator for being and admin user which i don't remember the name of.